### PR TITLE
AAC vent pressure control + fixes AAC vent & scrubber interaction

### PR DESCRIPTION
**Pull request by PJB3005**
fixes AAC interaction with vents & scrubbers.
Adds vent pressure control to the AAC.

### CODE REVIEW

File: code/modules/atmos_automation/implementation/vent_pump.dm [hunk 1]
```diff
@@ -88,4 +88,202 @@
...
+/datum/automation/set_vent_pump_pressure//controls the internal/external pressure bounds of a vent pump.
+	name = "Vent pump: Pressure settings"
+
+	var/vent_pump = null
+	var/intpressureout = 0//these 2 are for DP vents, if it's a unary vent you're sending to it will take intpressureout as var
+	var/intpressurein = 0
+	var/extpressure = 0
+	var/mode = 0//0 for unary vents, 1 for DP vents.
+
+	Export()
```
Comment (ComicIronic) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: Pathing should be absolute.

File: code/modules/atmos_automation/implementation/vent_pump.dm [hunk 2]
```diff
@@ -88,4 +88,202 @@
...
+	var/mode = 0//1 for DP vent, 0 for unary vent
+/*
+checks bitflags
+1 = external
+2 = internal in (regular internal for unaries)
+4 = internal out (ignored by unaries)
+*/
+
+
+	Export()
```
Comment (ComicIronic) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: Pathing again.

File: code/modules/atmos_automation/implementation/vent_pump.dm (hunk 1 above)
Comment (PJB3005) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: The pathing in all the other automation defines was relative too, I just followed it like it was

File: code/modules/atmos_automation/implementation/vent_pump.dm (hunk 2 above)
Comment (PJB3005) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: The pathing in all the other automation defines was relative too, I just followed it like it was

**Comment by Angelite191** [comment]
BAAAAAAAAAAASED! Next add the ability to link volume pumps too.

**Comment by PJB3005** [comment]
just use a valve infront of the pump, Angelite191, unless you're talking about pressure

**Comment by Angelite191** [comment]
I do that already, but it would be far more convenient to just link the volume pump.

**Comment by Angelite191** [comment]
Oh, I realize I sound pushy in my request. My apologies.

File: code/modules/atmos_automation/implementation/vent_pump.dm (hunk 1 above)
Comment (ComicIronic) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: Ideally they should all be absolute pathed.
One issue that I can see arising from this is a bunch of similarly written code being incredibly hard to read and easy to break.

Comment (d3athrow) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: >The pathing in all the other automation defines was relative too, I just followed it like it was

@PJB3005 What we do when we encounter relative pathing, color macro usage, etc in a file we're working in is to make it conform to current standards of span classes and absolute paths.

Comment (PJB3005) [inline code/modules/atmos_automation/implementation/vent_pump.dm]: i'll un relatify it then

**Comment by d3athrow** [comment]
Thanks bud.

**Pull request merged**